doc/source/indexing.rst

```diff
@@ -79,6 +79,7 @@ of multi-axis indexing.
   - A slice object with labels ``'a':'f'``, (note that contrary to usual python
     slices, **both** the start and the stop are included!)
   - A boolean array
+  - A ``callable``
```

Comment:
Can you add here a bit more specifics. eg "function with one argument (the calling dataframe/series) and that returns valid output for indexing (one of the above)"